# Enable logging the rate of processing with frames per second
print_fps: true

# Minimum height distance from camera for filtered point cloud (m)
z_axis_min: -0.23

# Maximum height distance from camera for filtered point cloud (m)
z_axis_max: 0.1

# Minimum forward distance from camera for filtered point cloud (m)
x_axis_min: 0.4

# Maximum forward distance from camera for filtered point cloud (m)
x_axis_max: 5

# Point cloud down sample resolution (m)
voxel_grid_leaf_size: 0.01

# Minimum allowable number of points in point cloud cluster
cluster_size_min: 1

# Maximum allowable number of points in point cloud cluster
cluster_size_max: 750

# Region growing parameters
smoothness_threshold: 5.0 # deg
curvature_threshold: 1000000.0

# Threshold angle to check for vertical surfaces (deg)
verticality_threshold: 20.0

# Length of pallet (m)
pallet_widths: [1.0]

# Tolerance for pallet length (m)
length_tolerance: 0.02

# Tolerance for angle difference between normals of pallet surfaces (deg)
angle_tolerance: 5.0

# Tolerance for midpoint location of the pallet (m)
midpoint_tolerance: 0.05

# Maximum height difference between pallet edges (m)
max_z_diff: 0.05

# Dock position offset from pallet position
x_offset: 0.9 # m
y_offset: 0.0 # m
yaw_offset: 180 # deg

# Dock position tolerance
goal_xy_tolerance: 0.025 # m
goal_yaw_tolerance: 5 # deg

# Distance tolerance to filter out outlier dock position
inlier_dist: 0.2 # m
inlier_yaw: 10 # deg

# Number of frames tracked to determine the average dock position
frames_tracked: 5

# Target frame to transform the point cloud to
target_frame: ""
